File: tests/fixtures/php/complex_methods.php
<?php

function process_order(int $status, int $priority, float $amount, float $discount, float $tax): float {
    $result = 0;
    if ($status == 1) {
        if ($priority > 5) {
            $result = $amount * 2;
        } elseif ($priority > 3) {
            $result = $amount;
        } else {
            $result = $amount / 2;
        }
    } elseif ($status == 2) {
        if ($discount > 0) {
            $result = $amount - $discount;
        }
    } elseif ($status == 3) {
        switch ($priority) {
            case 1:
                $result = $amount + $tax;
                break;
            case 2:
                $result = $amount - $tax;
                break;
            case 3:
                $result = $amount;
                break;
        }
    }
    if ($result < 0) {
        $result = 0;
    }
    return $result;
}

function simple_helper(int $x): int {
    return $x + 1;
}

File: tests/fixtures/lua/complex_methods.lua
function processOrder(status, priority, amount, discount, tax)
    local result = 0
    if status == 1 then
        if priority > 5 then
            result = amount * 2
        elseif priority > 3 then
            result = amount
        else
            result = amount / 2
        end
    elseif status == 2 then
        if discount > 0 then
            result = amount - discount
        end
    elseif status == 3 then
        if priority == 1 then
            result = amount + tax
        elseif priority == 2 then
            result = amount - tax
        elseif priority == 3 then
            result = amount
        else
            result = 0
        end
    end
    if result < 0 then
        result = 0
    end
    return result
end

function simpleHelper(x)
    return x + 1
end
